File: lecture02/03_pbbmatched/template.cpp
// pbbmatched.cpp

#include <iomanip>
#include <iostream>
#include <stack>
#include <string>

using namespace std;

// Functions

bool is_pbbmatched(const string &s) {
    // TODO: Process string s using a stack to determine if the symbols are balanced
    return false;
}

// Main execution

int main(int argc, char *argv[]) {
    string line;

    while (getline(cin, line)) {
    	cout << setw(10) << line << ": "<< (is_pbbmatched(line) ? "Yes" : "No") << endl;
    }

    return 0;
}

File: lecture02/03_pbbmatched/template.py
#!/usr/bin/env python3

import sys

# Functions

LEFT_PBB  = ('(', '[', '{')
RIGHT_PBB = (')', ']', '}')

def is_pbbmatched(s):
    # TODO: Process string s using a stack to determine if the symbols are balanced 
    return False

# Main execution

if __name__ == '__main__':
    for line in sys.stdin:
        line   = line.rstrip()
        result = 'Yes' if is_pbbmatched(line) else 'No'
        print('{:>10}: {}'.format(line, result))
